api/src/main/java/org/openmrs/api/impl/VisitServiceImpl.java

```diff
@@ -379,6 +380,120 @@ public void stopVisits(Date maximumStartDate) {
 		}
 	}
 	
+	/**
+	 * @see org.openmrs.api.VisitService#isSuitableVisit(Visit, Location, Date)
+	 */
+	@Override
+	public boolean isSuitableVisit(Visit visit, Location location, Date when) {
+		if (OpenmrsUtil.compare(when, visit.getStartDatetime()) < 0) {
+			return false;
+		}
+		if (OpenmrsUtil.compareWithNullAsLatest(when, visit.getStopDatetime()) > 0) {
+			return false;
+		}
+		return isSameOrAncestor(visit.getLocation(), location);
+	}
+	
+	@Override
+	@Transactional
+	/**
+	 * @see org.openmrs.api.VisitService#ensureVisit(Patient, Date, Location)
+	 */
+	public Visit ensureVisit(Patient patient, Date visitTime, Location department) {
```

Comment:
Arrange this statements. Isnt it supposed to be 

```suggestion
	/**
	 * @see org.openmrs.api.VisitService#ensureVisit(Patient, Date, Location)
	 */
    @Override
	@Transactional
	public Visit ensureVisit(Patient patient, Date visitTime, Location department) {
```